File: src/main/java/com/joraph/JoraphException.java
package com.joraph;

/**
 * Base exceptions used within the Joraph library.
 */
@SuppressWarnings("serial")
public class JoraphException
		extends RuntimeException {

	/**
	 * Creates a new instance of JoraphException.
	 * @param message exception message
	 */
    public JoraphException(String message) {
        super(message);
    }

	/**
	 * Creates a new instance of JoraphException.
	 * @param message exception message
	 * @param cause the root throwable
	 */
    public JoraphException(String message, Throwable cause) {
        super(message, cause);
    }

	/**
	 * Creates a new instance of JoraphException.
	 * @param cause the root throwable
	 */
    public JoraphException(Throwable cause) {
        super(cause);
    }

}

File: src/main/kotlin/com/joraph/JoraphException.kt
package com.joraph

/**
 * Base exceptions used within the Joraph library.
 */
open class JoraphException : RuntimeException {
    /**
     * Creates a new instance of JoraphException.
     * @param message exception message
     */
    constructor(message: String?) : super(message) {}

    /**
     * Creates a new instance of JoraphException.
     * @param message exception message
     * @param cause the root throwable
     */
    constructor(message: String?, cause: Throwable?) : super(message, cause) {}

    /**
     * Creates a new instance of JoraphException.
     * @param cause the root throwable
     */
    constructor(cause: Throwable?) : super(cause) {}
}
